src/drivers/indexeddb.js

```diff
@@ -947,36 +951,57 @@ function dropInstance(options, callback) {
     if (!options.name) {
         promise = Promise.reject('Invalid arguments');
     } else {
-        var dbPromise =
-            options.name === currentConfig.name && self._dbInfo.db
-                ? Promise.resolve(self._dbInfo.db)
-                : _getOriginalConnection(options);
+        const isCurrentDb =
+            options.name === currentConfig.name && self._dbInfo.db;
+
+        const dbPromise = isCurrentDb
+            ? Promise.resolve(self._dbInfo.db)
+            : _getOriginalConnection(options).then(function(db) {
+                  const dbContext = dbContexts[options.name];
+                  const forages = dbContext.forages;
+                  dbContext.db = db;
+                  for (var i = 0; i < forages.length; i++) {
+                      forages[i]._dbInfo.db = db;
+                  }
+                  return db;
+              });
 
         if (!options.storeName) {
-            promise = dbPromise.then(function() {
+            promise = dbPromise.then(function(db) {
                 _deferReadiness(options);
 
-                var dbContext = dbContexts[options.name];
-                var forages = dbContext.forages;
+                const dbContext = dbContexts[options.name];
+                const forages = dbContext.forages;
 
+                db.close();
                 for (var i = 0; i < forages.length; i++) {
                     var forage = forages[i];
-                    if (forage._dbInfo.db) {
-                        forage._dbInfo.db.close();
-                        forage._dbInfo.db = null;
-                    }
+                    forage._dbInfo.db = null;
                 }
 
                 var dropDBPromise = new Promise(function(resolve, reject) {
                     var req = idb.deleteDatabase(options.name);
 
-                    req.onerror = req.onblocked = reject;
+                    req.onerror = req.onblocked = e => {
```

Comment:
I'd like an eslint rule on this in general, but I think especially here as we already have a few `=` on the line, it'd be nice to do `(e) => {` instead of `e => {` because I find the parenthesis help indicate a function a lot better.

Also, could we use `err` or `error` here instead? I know `e` usually means error but I dislike single-lettle variables 😄

Reply:
Changed that, added an eslint rule, but in the end prettier didn't like it and I couldn't find a way to change its mind :(